### DISCUSSION TITLE
Floor vs Round

### DISCUSSION THREAD

**Discussion by fredvs** [mse-org/mseide-msegui#73] [General]
Following that issue: https://github.com/bgrabitmap/bgrabitmap/issues/238
Do you agree to change all **round()** with **floor()** in MSEgui code?

**Comment by rchastain** [comment]
To solve which problem?

**Comment by rchastain** [comment]
FYI, I see this in _mseplitter.pas:_

```
./lib/common/widgets/msesplitter.pas:447:{$define useround}
./lib/common/widgets/msesplitter.pas:903:  newsize.cx:= {$ifdef useround}round{$else}floor{$endif}(refsize.cx * fhsizeprop);
./lib/common/widgets/msesplitter.pas:909:  newsize.cy:= {$ifdef useround}round{$else}floor{$endif}(refsize.cy * fvsizeprop);
./lib/common/widgets/msesplitter.pas:919:  offset.x:= offset.x + {$ifdef useround}round{$else}floor{$endif}(fhprop * refsize.cx) - parentclientpos.x -
./lib/common/widgets/msesplitter.pas:927:  offset.y:= offset.y + {$ifdef useround}round{$else}floor{$endif}(fvprop * refsize.cy) - parentclientpos.y -
```

**Comment by fredvs** [comment]
Hello Roland.

Ho, ok, Martin did it already.

>To solve which problem?

Maybe same adjustment like in BGRABitmap issue.

In msesplitter.pas, you found {$define useround}, do you have a idea why floor() should be used instead?

Maybe it is a tempete dans un verre d'eau and MSEgui did solve it already.

I apologize for the noise.

Ftre;D

**Reply by rchastain** [reply]
No need for apologies. We can discuss the question. But I see no reason to modify something for now.

**Reply by fredvs** [reply]
Yes, it is why I love to work on a project where discussion is possible!
(And maybe I changed too fast all the round() with floor() in uos project).
